Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class Lesson4
+    {
+        public static void Demo()
+        {
+            int numberOfIndividuals = PromptInt("Please enter the number of individuals you would like to calculate the BMI of: ");
+            while (numberOfIndividuals > 0)
+            {
+                PromptForPersonalData();
+
+                numberOfIndividuals--;
+            }
+        }
+
+        static void PromptForPersonalData()
+        {
+            string firstName = PromptString("Please enter the individual's first name: ");
+            string lastName = PromptString("Please enter the individual's last name: ");
+            int age = PromptInt("Please enter the individual's age: ");
+            float weight = PromptFloat("Please enter the individual's weight (in kg, example: 70.5): ");
+            float height = PromptFloat("Please enter the individual's height (in meters, example: 1.8): ");
+            float bodyMassIndex = CalculateBMI(weight, height);
+
+            Console.WriteLine($"{firstName} {lastName} is {age} years old, their weight is {weight} kg and their height is {height} m.");
+            Console.WriteLine($"{firstName} {lastName}'s BMI is: {bodyMassIndex}");
+        }
+
+
+        public static string PromptString(string message)
+        {
+            Console.WriteLine(message);
+            return ValidateString(Console.ReadLine());
+        }
+
+        public static int PromptInt(string message)
+        {
+            Console.WriteLine(message);
+            return ValidateInt(Console.ReadLine());
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.WriteLine(message);
+            return ValidateFloat(Console.ReadLine());
+        }
+
+        public static float CalculateBMI(float weight, float height)
+        {
+            if (weight <= 0 || height <= 0)
```

Comment:
And I think this whole block of code could go into a function of its' own:
```cs
private static bool IsValidBmi(float weight, float height)
{
  // do validation and print
  return true if valid, else invalid;
}
```

Reply:
And then:
```cs
if(!IsValidBmi(weight, height))
{
  return -1;
}
return bmiFormula...
```

Reply:
New function has been added and responsiblities have been separated as described in the previous comment above.